lib/internal/fs/cp/cp-sync.js

```diff
@@ -199,7 +199,7 @@ function onLink(destStat, src, dest, verbatimSymlinks) {
   if (!isAbsolute(resolvedDest)) {
     resolvedDest = resolve(dirname(dest), resolvedDest);
   }
-  if (isSrcSubdir(resolvedSrc, resolvedDest)) {
+  if (fsBinding.internalModuleStat(src) === 1 && isSrcSubdir(resolvedSrc, resolvedDest)) {
```

Comment:
could we maybe use a temporary variable to make the code more understandable? :slightly_smiling_face: 
```suggestion
  const srcIsDir = fsBinding.internalModuleStat(src) === 1;
  if (srcIsDir && isSrcSubdir(resolvedSrc, resolvedDest)) {
```